### PR TITLE
Feat: vercel integration

### PR DESCRIPTION
**Pull request by nimish-ks**
## :mag: Overview
Add support for syncing secrets to Vercel environments. This integration allows users to sync their Phase secrets to Vercel projects across different environments (production, preview, development) with configurable secret types.

## :bulb: Proposed Changes
- Add Vercel integration backend with support for:
  - Credential validation
  - Project listing
  - Secret syncing with different visibility types
- Add frontend components for Vercel sync configuration
- Add GraphQL mutations and queries for Vercel integration
- Add new service type in ServiceConfig

## :framed_picture: Screenshots or Demo

Screenshot From 2024-11-12 11-56-52
Screenshot From 2024-11-12 11-57-13
Screenshot From 2024-11-12 11-57-46
Screenshot From 2024-11-12 11-58-44

### Added
- New Vercel integration for secret syncing
- Support for different Vercel environments (production, preview, development, all)
- Multiple secret visibility options (plain, encrypted, sensitive)
- Vercel project selection with search functionality

## Docs
Draft: https://github.com/phasehq/docs/pull/98

### :dart: Reviewer Focus
Please focus on:
1. Credential validation flow
   - Test with valid Vercel tokens
   - Test with invalid/expired tokens
   - Test with non-Vercel credentials
2. Secret type handling
   - Verify plain text secrets are visible in Vercel UI
   - Verify encrypted secrets are hidden but accessible
   - Verify sensitive secrets are completely hidden
3. Environment targeting
   - Test syncing to specific environments (prod/preview/dev)
   - Test syncing to all environments
   - Verify environment-specific secrets are isolated

### :sparkles: How to Test the Changes Locally
1. Add Vercel credentials in Phase:
   - Get a Vercel access token from https://vercel.com/account/tokens
   - Add credentials in Phase under Integration Settings
2. Create a new sync:
   - Select a Phase environment
   - Choose your Vercel project
   - Select target environment and secret type
3. Add secrets and verify they appear in your Vercel project settings

### :green_heart: Did You...
- [x] Ensure linting passes (code style checks)
- [ ] Update dependencies and lockfiles (if required)
- [x] Regenerate graphql schema and types (if required)
- [x] Verify the app builds locally
- [ ] Manually test the changes on different browsers/devices

### CODE REVIEW

File: frontend/components/syncing/Vercel/CreateVercelSync.tsx [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+import { 
+  EnvironmentType, 
+  ProviderCredentialsType,
+  VercelProjectType 
+} from '@/apollo/graphql'
+import { Combobox, RadioGroup, Transition } from '@headlessui/react'
+import clsx from 'clsx'
+import { FaAngleDoubleDown, FaChevronDown, FaCircle, FaDotCircle } from 'react-icons/fa'
+import { toast } from 'react-toastify'
+import { TbBrandVercel } from 'react-icons/tb'
```
Comment (rohan-chaturvedi) [inline frontend/components/syncing/Vercel/CreateVercelSync.tsx]:
```suggestion
import { SiVercel } from 'react-icons/si'
```

Lets stick with the icon from Simple Icons for consistency

File: frontend/components/syncing/Vercel/CreateVercelSync.tsx [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+    { value: 'plain', label: 'Plain Text' },
+    { value: 'encrypted', label: 'Encrypted' },
+    { value: 'sensitive', label: 'Sensitive' }
+  ]
+
+  return (
+    <div className="p-4 space-y-6">
+      <div>
+        <div className="text-2xl font-semibold text-black dark:text-white flex items-center gap-2">
+          <TbBrandVercel className="text-2xl" />
```
Comment (rohan-chaturvedi) [inline frontend/components/syncing/Vercel/CreateVercelSync.tsx]:
```suggestion
          <SiVercel className="text-2xl" />
```

File: backend/backend/graphene/queries/syncing.py [hunk 3]
```diff
@@ -162,6 +166,26 @@ def resolve_railway_projects(root, info, credential_id):
...
+    except Exception as ex:
+        raise GraphQLError(f"Error listing Vercel projects: {str(ex)}")
+
+def resolve_test_vercel_creds(root, info, credential_id):
+    """Resolver for testing Vercel credentials."""
+    try:
+        valid = test_vercel_creds(credential_id)
+        return valid
+    except Exception as ex:
+        raise GraphQLError(f"Error testing Vercel credentials: {str(ex)}")
```
Comment (rohan-chaturvedi) [inline backend/backend/graphene/queries/syncing.py]: This doesn't seem to be required, lets remove this resolver and the related import in `schema.py`

**Review by rohan-chaturvedi — CHANGES_REQUESTED** [review]
Awesome work! Seems to work flawlessly, Just a few minor suggestions.

File: backend/backend/graphene/queries/syncing.py (hunk 3 above)
Comment (nimish-ks) [reply backend/backend/graphene/queries/syncing.py]: Yeah, good call. Removed it. Regenerated graphql types.

**Pull request merged**